src/logic/calculate.js

```diff
@@ -0,0 +1,27 @@
+/* eslint-disable import/extensions */
+/* eslint-disable prefer-const */
+
+import operate from './operate.js';
+
+const calculate = (calc, name) => {
+  let { total, next, operation } = calc;
```

Comment:
Can we use `const` here instead? The `total, next, operation` doesn't need to be mutated directly, we can create new variable that would be mutated if needed. Eg: 
```js
const { total, next, operation } = calc;
let newTotal = total;
let newNext = next;

// Do something

return { total: newTotal, next: newNext, operation };
```

Note that this is only an example to can achieve the same result in other ways.